tests/parsers/test_EquationParser.py

```diff
@@ -0,0 +1,123 @@
+    test_name = "test_time_dependent_equation"
+    equation = '10*t'
+    expression = EquationParser(equation)
+    variables = ['t']
+
+    test_inputs = [[10],
+                   [20],
+                   [30.42]]
+    expected_results = [100, 200, 304.2]
+
+    run_all_values(variables, expression, test_name, equation,
+                    test_inputs, expected_results)
+
+def test_spatial_and_time_equation():
+    test_name = "test_spatial_and_time_equation"
+    equation = '(10.1*x + (15.6 / y) - z**2) / t'
+    expression = EquationParser(equation)
+    variables = ['x', 'z', 'y', 't']
+
+    test_inputs = [(1, 1, 1, 1),
+                   (5, 5, 5, 5),
+                   (1.2, 2.0, 0.2, 0.8)]
+    expected_results = [24.7, 5.724, 24.85]
+
+    run_all_values(variables, expression, test_name, equation,
+                    test_inputs, expected_results)
+
+def test_coupled_variables():
+    test_name = "test_coupled_variables"
+    equation = '(11.9*x + 3.6*z) * 13.2*temperature + 6*pressure'
+    expression = EquationParser(equation, 'temperature', 'pressure')
+    variables = ['x', 'z', 'temperature', 'pressure']
+
+    test_inputs = [(1, 1, 1, 1),
+                   (5, 5, 5, 5),
+                   (1.2, 2.0, 0.2, 0.8)]
+    expected_results = [210.6, 5145, 61.5072]
+
+    run_all_values(variables, expression, test_name, equation,
+                    test_inputs, expected_results)
+
+def test_unit_conversion():
+    test_name = "test_unit_conversion"
+    equation = '10.1 + 3.2*x'
+    expression = EquationParser(equation)
+    variables = ['x']
+
+    scale_factor = 1.8
+    shift_factor = 22
+    expression.performUnitConversion(scale_factor, shift_factor)
+
+    test_inputs = [[1],
+                   [5],
+                   [4.6]]
+    expected_results = [45.94, 68.98, 66.676]
+
+    run_all_values(variables, expression, test_name, equation,
+                    test_inputs, expected_results)
+
+def run_all_values(variables : list, expression, test_name, original_equation, test_inputs, expected_values):
+    number_of_variables = len(variables)
+    number_of_tests = len(test_inputs)
+
+    inputs_and_expected = [(test_inputs[i], expected_values[i]) for i in range(number_of_tests)]
+    for input, expected in inputs_and_expected:
+        input_args = {variables[i] : input[i] for i in range(number_of_variables)}
+        output_value = expression.evaluate(**input_args)
+        compare_results(test_name, original_equation, input,
+                        output_value, expected)
+
+def compare_results(test_name, equation, input_values, output_value, expected_value):
+    if withinRoundingPrecision(expected_value, output_value):
+        return True
+    else:
+        raise Exception("ERROR in test " + test_name + " : " + \
+                         "Expected " + str(equation) + " @ " + str(input_values) + \
+                         "to give '"+str(expected_value)+"'. Got '"+str(output_value)+"' instead.")
+
```

Comment:
Typically, and `assert isclose` will suffice.  No need for the long exception raising.  An assertion will clue the programmer into where and what to check.  Also, the `assert` also has it's own string print if you want to use that.